### ISSUE TITLE
(Bug report) Today page does not work for 2021-11-20

### ISSUE THREAD

**Issue by agentydragon** [zadam/trilium#2359]
### Preflight Checklist

- [X] I have searched the [issue tracker](https://www.github.com/zadam/trilium/issues) for a bug report that matches the one I want to file, without success.

### Trilium Version

0.48.6

### What operating system are you using?

Ubuntu

### What is your setup?

Local + server sync

### Operating System Version

Pop!_OS 20.04

### Expected Behavior

Clicking "today" button should create new page for 2021-11-20 and navigate to it. Opening calendar and clicking on 2021-11-20 should create new page for the date and navigate to it.

### Actual Behavior

For some reason, on this particular day (2021-11-20), clicking the Today button, I'm taken on the page for 2021-11-09.

The 2021-11-09 page's `#dateNote` attribute is correctly set to 2021-11-09, *not* today's date 2021-11-20:
![image](https://user-images.githubusercontent.com/714892/142721769-d9736222-d49e-4b25-a923-df6d19d34fe6.png)

Those are the inherited attributes:
![image](https://user-images.githubusercontent.com/714892/142721822-8e4e7cda-a8e8-46d5-bae0-6c833c5c5ec2.png)

Clicking on the boldened today's date (2021-11-20) in calendar also navigates to 2021-11-09 for some reason.

Clicking on 2021-11-21 in calendar works just fine - creates the 2021-11-21 page and navigates to it:
![image](https://user-images.githubusercontent.com/714892/142721797-f612eeed-9c39-49d5-b5ba-81e3c386c45b.png)

**Comment by zadam** [comment]
Hi, interesting bug :D

Unfortunately, this seems to work for me. Does the issue persist if you restart the application?

**Comment by agentydragon** [comment]
Yes. Restarting Trilium client did not help. Did not yet try to restart
server.

**Comment by agentydragon** [comment]
Backend restart didn't help either.

**Comment by agentydragon** [comment]
Force full sync, Fill entity changes records, Find and fix consistency issues, Vacuum database, followed by restart of frontend didn't help.

![image](https://user-images.githubusercontent.com/714892/142735522-d36a2798-7afd-4d01-b940-b5c33491bba7.png)

**Comment by agentydragon** [comment]
This happens both on local client, and when I open the server's URL in a web browser. Happens the same both in mobile web frontend and "desktop web frontend".

**Comment by agentydragon** [comment]
I just created today's note manually and set `#dateNote=2021-11-20` on it. Now clicking the today button works as expected, so this is low-pri since the workaround works.

**Comment by agentydragon** [comment]
And today, it seems to work. So I think no super high priority on figuring this out. I'll be back if I see this ever again.

**Comment by agentydragon** [comment]
Encountered this again - attempting to open 2021-12-20 opens an existing page 2021-12-13 for me instead.

**Comment by zadam** [comment]
> In case it makes a difference, I use #datePattern="{isoDate} - {weekDay}" 

This was the missing piece which was visible on the screenshot, but I did not notice / put any importance on that.

Back in the very first versions of Trilium notes did not have labels and Journal tree was created simply as a tree of notes with a fixed format. Then the labels came but to transition from the previous non-labeled structure there was a heuristic which in case the labeled note was not found, it will try to find the note by name prefix. This then conflicted with "isoDate" template on 20th day in the month since all isoDate journal notes start with year like "2021".

So I fixed it by removing the transitioning code. I doubt this removal will affect anybody.

Thanks for helping to get to the bottom of this!
